benchmark/scripts/Benchmark_Driver

```diff
@@ -129,6 +129,12 @@ def get_current_git_branch(git_repo_path):
         ['git', '-C', git_repo_path, 'rev-parse',
          '--abbrev-ref', 'HEAD'], stderr=subprocess.STDOUT).strip()
 
+def get_git_head_ID(git_repo_path):
+    """Return the short identifier for the HEAD commit of the repo
+        `git_repo_path`"""
+    return subprocess.check_output(
+        ['git', '-C', git_repo_path, 'rev-parse',
+         '--short', 'HEAD'], stderr=subprocess.STDOUT).strip()
```

Comment:
I didn't want to make the log filename overly long. Currently we have:
`Benchmark_O-20170421004027.log`
my change makes it:
`Benchmark_O-20170422050121-077340abd7.log`
full hash would be IMHO unwieldy:
`Benchmark_O-20170422050121-f3b33f06d4c60f70afc591f2d261e7aac1c5d0ac.log`

In [SR-4675](https://bugs.swift.org/browse/SR-4675) "Report detailed build version from Benchmark drivers" I plan to report full hash and will add it into the report itself. So that will take care of that.

Also, there will be no conflicts with this - filename is still unique due to timestamp.